Library/Homebrew/readall.rb

```diff
@@ -35,28 +36,43 @@ def valid_aliases?(alias_dir, formula_dir)
     end
 
     def valid_formulae?(formulae)
-      failed = false
+      success = true
       formulae.each do |file|
         Formulary.factory(file)
       rescue Interrupt
         raise
       rescue Exception => e # rubocop:disable Lint/RescueException
         onoe "Invalid formula: #{file}"
-        puts e
-        failed = true
+        $stderr.puts e
+        success = false
       end
-      !failed
+      success
+    end
+
+    def valid_casks?(casks)
+      success = true
+      casks.each do |file|
+        Cask::CaskLoader.load(file)
+      rescue Interrupt
+        raise
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        onoe "Invalid cask: #{file}"
+        $stderr.puts e
+        success = false
+      end
+      success
     end
 
     def valid_tap?(tap, options = {})
-      failed = false
+      success = true
       if options[:aliases]
         valid_aliases = valid_aliases?(tap.alias_dir, tap.formula_dir)
-        failed = true unless valid_aliases
+        success = false if !valid_aliases
```

Comment:
Sorry, yeh, this should be `unless`

Reply:
Maybe use `success &= valid_aliases` for all of these?